integration/plugin/authz/main_test.go

```diff
@@ -0,0 +1,177 @@
+// +build !windows
+
+package authz
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/docker/docker/integration-cli/daemon"
+	"github.com/docker/docker/internal/test/environment"
+	"github.com/docker/docker/pkg/authorization"
+	"github.com/docker/docker/pkg/plugins"
+)
+
+var (
+	testEnv *environment.Execution
+	d       *daemon.Daemon
+	server  *httptest.Server
+)
+
+const dockerdBinary = "dockerd"
+
+func TestMain(m *testing.M) {
+	var err error
+	testEnv, err = environment.New()
+	if err != nil {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	testEnv.Print()
+	setupSuite()
+	exitCode := m.Run()
+	teardownSuite()
+
+	os.Exit(exitCode)
+}
+
+func setupTest(t *testing.T) func() {
+	environment.ProtectImages(t, testEnv)
```

Comment:
We merged a PR recently that added a ProtectAll() I think that should be used now. We should unexport the rest of these Protect*() method so it's more obvious.

Reply:
If we used `ProtectAll`, wouldn't that negate the use of `testEnv.Clean`? I think that containers and plugins should probably be cleaned up after these tests, no?

Reply:
`ProtectAll()` is maybe a misleading name. All it does is get a list of all the existing objects, so that `Clean()` doesn't remove them later. This is important for testing EE because it has system containers that it needs to keep running.

`Clean()` will still remove anything created by the tests.